engine/tx_broadcast.go

```diff
@@ -0,0 +1,46 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chainstate"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+)
+
+func broadcastTransaction(ctx context.Context, tx *Transaction) error {
+	client := tx.Client()
+	chainstateSrv := client.Chainstate()
+
+	defer recoverAndLog(tx.Client().Logger())
+
+	unlock, err := newWriteLock(
+		ctx, fmt.Sprintf(lockKeyProcessBroadcastTx, tx.GetID()), client.Cachestore(),
+	)
+	defer unlock()
+	if err != nil {
+		return err
+	}
+
+	txHex := tx.Hex
+	hexFormat := chainstate.RawTx
+	if chainstateSrv.SupportedBroadcastFormats().Contains(chainstate.Ef) {
+		// try to convert to EF, with rawRx as a fallback
+		efHex, ok := ToEfHex(ctx, tx, client)
+		if ok {
+			txHex = efHex
+			hexFormat = chainstate.Ef
+		}
+	}
+
+	br := chainstateSrv.Broadcast(ctx, tx.ID, txHex, hexFormat, defaultBroadcastTimeout)
+
+	if br.Failure != nil {
```

Comment:
again i would log the failure 
```if br.Failure != nil {
		tx.Client().Logger().Error().Str("txID", tx.ID).Err(br.Failure.Error).Msg("transaction broadcast failed")
		if br.Failure.InvalidTx {
			return spverrors.ErrBroadcastRejectedTransaction.Wrap(br.Failure.Error)
		}
		return br.Failure.Error
	}
```

Reply:
This "Failure" doesn't mean that something went wrong in the `spv-wallet`.
Maybe it is an invalid transaction or ARC simply doesn't know (yet) anything about this transaction. 
So, it's a part of "normal" flow and I thing the logging here is not necessary - moreover, for some cases it is actually logged, but at higher level.

Reply:
so then the naming is misleading, and you should call it `invalid` not sure but anything what is marked with words invalid or failure or error in my opinion should be logged as close as possible to the check / processing tx

Reply:
The name is from `chainstate` which will be reworked soon (as I said above ☝️: `SPV-1061`)

However I do agree that the name is misleading - I'll try to implement/name it better in this future task. 
For now, it's out of scope of this task.